src/theory/builtin/theory_builtin_rewriter.h

```diff
@@ -62,6 +62,16 @@ class TheoryBuiltinRewriter {
   /** recursive helper for getArrayRepresentationForLambda */
   static Node getArrayRepresentationForLambdaRec( TNode n, bool reqConst, TypeNode retType );
 public:
+  /** Get function type for array type
+  * This returns the function type of terms returned by the function getLambdaForArrayRepresentation( t, bvl ),
+  * where t.getType()=atn.
+  */
+  static TypeNode getFunctionTypeForArrayType( TypeNode atn, Node bvl );
```

Comment:
Why are these public and static in TheoryBuiltinRewriter? Who needs to access them? Do these need special access to anything in TheoryBuiltinRewriter?

Reply:
They are useful in the conversion between functions and arrays.  FunctionEnumerator is an example of something that needs them.

Reply:
From your description it sounds like these are unrelated to TheoryBuiltinRewriter then. Consider making these free functions.

Reply:
These functions should be in the same place as getLambdaForArrayRepresentation and getArrayRepresentationForLambda, which are used in the builtin rewriter.  At the moment I think this is the best place for these 4 functions.